src/ui/ui-storyEditor.js

```diff
@@ -224,6 +237,49 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
                         }
                     }
                 }
+            },
+            reorderer: {
+                type: "fluid.reorderList",
+                container: "{storyEditor}.dom.storyEditorContent",
+                createOnEvent: "{storyEditor}.events.onEditorTemplateRendered",
+                options: {
+                    disableWrap: true,
+                    styles: {
+                        avatar: "sjrk-st-reorderer-block-avatar",
+                        dropMarker: "sjrk-st-reorderer-block-dropmarker"
+                    },
+                    selectors: {
+                        movables: ".sjrkc-dynamic-view-component",
+                        selectables: ".sjrkc-dynamic-view-component",
+                        dropTargets: ".sjrkc-dynamic-view-component"
+                    },
+                    invokers: {
+                        reorderBlock: {
+                            funcName: "sjrk.storyTelling.ui.storyEditor.reorderBlock",
+                            args: ["{that}", "{arguments}.0", "{arguments}.1"] // blockUi, direction
+                        },
+                        reorderBlockDown: {
+                            func: "{that}.reorderBlock",
+                            args: ["{arguments}.0", fluid.direction.DOWN]
+                        },
+                        reorderBlockUp: {
+                            func: "{that}.reorderBlock",
+                            args: ["{arguments}.0", fluid.direction.UP]
+                        }
+                    },
+                    listeners: {
+                        // We should be refreshing the reorderer on each new block
+                        // being added, but we can't reliably know when a block
+                        // has finished being rendered. It would also be possible
```

Comment:
@jobara and I were exploring this, we even attempted to incorporate a `mutationObserver` into the `storyEditor` before that became too unwieldy, but we believe the issue boils down to [`gpii.handlebars.renderer`'s use](https://github.com/GPII/gpii-handlebars/blob/master/src/js/client/renderer.js#L15-L23) of jQuery functions to affect markup changes. In this case, we're using [`$.html()`](https://api.jquery.com/html/).

Since jQuery doesn't afford any way to know that it's done its thing, we can't reliably know when a newly-added blockUi is ready to be fiddled with, and therefore when it's safe to refresh the reorderer (this becomes an issue once a `grabHandle` is specified for a `movable`, something to do with the reorderer's `labeller` component, as I recall).

I've messaged @the-t-in-rtf to discuss how we might solve this issue, but I hadn't provided him with much detail and our workaround made it a less pressing matter.

Reply:
I can't quite follow your explanation - $.html() is synchronous so it has "done it's thing" immediately?

Reply:
I believe I misunderstood the root cause of the issue. In the case that we call the `refresh` invoker after adding a new block, the problem would arise when we attempt to add many blocks very quickly, either on loading an autosave or simply clicking the add button very rapidly. The reorderer will attempt to refresh itself, and upon coming into contact with a block that has a container added but not actually rendered, we'll get the error.

@jobara and I have discussed implementing a kind of event queue to manage adding blocks one at a time and waiting until the queue is empty to call refresh